core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -115,20 +118,37 @@ private[spark] class CoarseCookSchedulerBackend(
   }
 
   var runningJobUUIDs = Set[UUID]()
+  var abortedJobIds = Set[UUID]()
+
+  private var jobLimitOption: Option[Int] = None
 
   val jobListener = new CJobListener {
     // These are called serially so don't need to worry about race conditions
     def onStatusUpdate(job : Job) {
-      if (job.getStatus == Job.Status.COMPLETED && !job.isSuccess) {
+      val isCompleted = job.getStatus == Job.Status.COMPLETED
+      val uuid = job.getUUID
+      val isAborted = abortedJobIds.contains(uuid)
+
+      if (isCompleted) {
         totalCoresRequested -= job.getCpus().toInt
-        totalFailures += 1
-        logWarning(s"Job ${job.getUUID} has died. Failure ($totalFailures/$maxFailures)")
-        runningJobUUIDs = runningJobUUIDs - job.getUUID
-        if (totalFailures >= maxFailures) {
-          // TODO should we abort the outstanding tasks now?
-          logError(s"We have exceeded our maximum failures ($maxFailures)" +
-                    "and will not relaunch any more tasks")
-        } else requestRemainingCores()
+        abortedJobIds = abortedJobIds - uuid
+      }
+
+      if (isCompleted && !job.isSuccess) {
+        if (!isAborted) {
+          totalFailures += 1
+          logWarning(s"Job ${job.getUUID} has died. Failure ($totalFailures/$maxFailures)")
+
+          if (totalFailures >= maxFailures) {
+            // TODO should we abort the outstanding tasks now?
+            logError(s"We have exceeded our maximum failures ($maxFailures)" +
+              "and will not relaunch any more tasks")
+          }
+        }
+
+        runningJobUUIDs = runningJobUUIDs - uuid
```

Comment:
You're right, I changed it (along with a couple other fixes). More info in my update on basecamp :)